kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/WifiInterfaceState.java

```diff
@@ -34,16 +34,33 @@
      */
     public WifiInterfaceState(String interfaceName, WifiMode wifiMode) throws KuraException {
         super(NetInterfaceType.WIFI, interfaceName);
-        if (WifiMode.MASTER.equals(wifiMode)) {
-            if (this.m_link) {
-                boolean isHostapdRunning = HostapdManager.isRunning(interfaceName);
-                boolean isIfaceInApMode = WifiMode.MASTER.equals(LinuxNetworkUtil.getWifiMode(interfaceName));
-                if (!isHostapdRunning || !isIfaceInApMode) {
-                    s_logger.warn("WifiInterfaceState() :: !! Link is down for the " + interfaceName
-                            + " interface. isHostapdRunning? " + isHostapdRunning + " isIfaceInApMode? "
-                            + isIfaceInApMode);
-                    this.m_link = false;
-                }
+        setWifiLinkState(interfaceName, wifiMode);
+    }
+
+    /**
+     * WifiInterfaceState
+     *
+     * @param interfaceName
+     *            - interface name as {@link String}
+     * @param wifiMode
+     *            - configured wifi mode as {@link WifiMode}
+     * @param isL2OnlyInterface
+     *            - is Layer 2 only interface
+     * @throws KuraException
+     */
+    public WifiInterfaceState(String interfaceName, WifiMode wifiMode, boolean isL2OnlyInterface) throws KuraException {
```

Comment:
Doesn't this change imply an update of the exported package in manifest file?

Reply:
Made changes to the MANIFEST.MF